src/sentry/api/endpoints/project_dynamic_sampling.py

```diff
@@ -0,0 +1,194 @@
+import math
+from datetime import timedelta
+
+from django.utils import timezone
+from rest_framework import status
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry import features
+from sentry.api.bases.project import ProjectEndpoint, ProjectPermission
+from sentry.models import Project
+from sentry.snuba import discover
+from sentry.utils.dates import parse_stats_period
+
+
+def percentile_fn(data, percentile):
+    """
+    Returns the nth percentile from a sorted list
+
+    :param percentile: A value between 0 and 1
+    :param data: Sorted list of values
+    """
+    if len(data) == 0:
+        return None
+    pecentile_idx = len(data) * percentile
+    if pecentile_idx.is_integer():
+        return data[int(pecentile_idx)]
+    else:
+        return data[int(math.ceil(pecentile_idx)) - 1]
+
+
+class DynamicSamplingPermission(ProjectPermission):
+    # ToDo(ahmed): Revisit the permission level for Dynamic Sampling once the requirements are
+    #  better defined
+    scope_map = {"GET": ["project:write", "project:admin"]}
+
+
+class ProjectDynamicSamplingDistributionEndpoint(ProjectEndpoint):
+    private = True
+    permission_classes = (DynamicSamplingPermission,)
+
+    @staticmethod
+    def _get_sample_rates_data(data):
+        distribution_functions = {
+            "min": lambda lst: min(lst, default=None),
+            "max": lambda lst: max(lst, default=None),
+            "mean": lambda lst: sum(lst) / len(lst) if len(lst) > 0 else None,
+            "p50": lambda lst: percentile_fn(data, 0.5),
+            "p90": lambda lst: percentile_fn(data, 0.9),
+            "p95": lambda lst: percentile_fn(data, 0.95),
+            "p99": lambda lst: percentile_fn(data, 0.99),
+        }
+        return {key: func(data) for key, func in distribution_functions.items()}
```

Comment:
```suggestion
        return {
            "min": min(data, default=None),
            "max": max(data, default=None),
            "mean": sum(data) / len(data) if len(data) > 0 else None,
            "p50": percentile_fn(data, 0.5),
            "p90": percentile_fn(data, 0.9),
            "p95": percentile_fn(data, 0.95),
            "p99": percentile_fn(data, 0.99),
        }
```

if you want to stick with your solution, in the percentiles lambdas you're not actually using the `lst` parameter